core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -431,12 +438,41 @@ private static boolean checkPosOccurrenceParamValues(int position,
         throw RESOURCE.invalidIntegerInputForRegexpFunctions(Integer.toString(occurrencePosition),
             "occurrence_position", methodName).ex();
       }
-      if (position <= value.length()) {
+      if (position > value.length()) {
         return false;
       }
       return true;
```

Comment:
Looking at the code smells, should we rewrite this as `return position <= value.length()` (maybe I did my logic table wrong but hopefully you get what I'm saying: Squash the `if` into a single return)

Reply:
Good point, Tanner. It vindicates my philosophy that method's name and javadoc should say what the method does and returns.

That a method is named `checkXxx` and returns `boolean` (or throws) is a code smell. Note that the other `checkXxx` methods are `void`. As soon as a method's return changes from `void` to `boolean` you should rename it and re-write its javadoc.